pkg/storage/store.go

```diff
@@ -3091,8 +3091,9 @@ func sendSnapshot(
 	// NB: wait for EOF which ensures that all processing on the server side has
 	// completed (such as defers that might be run after the previous message was
 	// received).
-	if _, err := stream.Recv(); err != io.EOF {
-		return errors.Errorf("range=%s: expected EOF, got %v", header.RangeDescriptor.RangeID, err)
+	if resp2, err := stream.Recv(); err != io.EOF {
```

Comment:
@vivekmenezes We don't loop because there is a protocol between the snapshot sender and the server that requires an EOF after ERROR or APPLIED is sent. If that protocol is violated, it seems better to break quickly (which not looping will do) than to forge ahead.